cron/dump_clean_daily

```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env bash
+
+# Performs a filtered dump of the measurements table.
+#
+# The filtering is the same as that used by the iOS app.
+#
+# This differs from dump_measurements, which does not filter data at all.
+# The schema returned is also different; redundant/unused columns have been eliminated, and id/user_id finally included.
+# The ORDER BY clause was removed for performance.
+
+set -euo pipefail
+
+source cron_env.sh
+
+cd ../public/system
+
+psql -f "${CRON_DIR}/dump_clean_daily.sql"
+
+tar -czf dump_clean_daily.tar.gz dump_clean_daily-out.csv
+mv dump_clean_daily-out.csv dump_clean_daily.csv
+
+# Now this is available as https://api.safecast.org/system/dump_clean_daily.tar.gz
```

Comment:
Is this sufficient for our users' needs? Or should we name these files differently for each day, and retain a few days of dumps? For instance `dump_clean_daily_2020-07-01.tar.gz`

Reply:
Not sure but I'm open for any suggestions

Reply:
Here's an example of how you could do this in bash. The below code sorts a list of archive names that are older than a cutoff time, and creates a command to delete them. This could be changed to work with filenames.

Another option would be to use `find` to look for files created before a certain day. However, the create date could be misleading.

```
cutoff_date="backup-$(/bin/date -u --iso-8601=seconds -d '-3 months')"
archives="$(/usr/bin/tarsnap --list-archives | /usr/bin/sort)"
cutoff_archives=""
cutoff_command="/usr/bin/tarsnap --humanize-numbers -d"

while IFS=' ' read -r archive; do
    cutoff_archive=$(echo -e "${cutoff_date}\n${archive}" \
            | /usr/bin/sort \
            | /usr/bin/head -n 1 \
            | /bin/grep -v -F "$cutoff_date" || true)
    if [ -z "$cutoff_archive" ]; then
        continue
    fi
    if [ -z "$cutoff_archives" ]; then
        cutoff_archives="${cutoff_archive}"
    else
        cutoff_archives="${cutoff_archives}\n${cutoff_archive}"
    fi
    cutoff_command="${cutoff_command} -f ${cutoff_archive}"
done <<< "$archives"
```

Reply:
Thank you very much but it might take time to me to understand what I’m looking at, haha. I’m trying to understand what is the task. Do we want to have a list of dumps back to three months? What is our goal? :)

Reply:
I added an issue